library/debug.c

```diff
@@ -410,4 +410,43 @@ void mbedtls_debug_printf_ecdh( const mbedtls_ssl_context *ssl, int level,
 }
 #endif /* MBEDTLS_ECDH_C */
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_SESSION_TICKETS)
+void mbedtls_debug_print_ticket_flags( const mbedtls_ssl_context *ssl, int level,
+                      const char *file, int line,
+                      mbedtls_ssl_tls13_ticket_flags flag )
+{
+    char str[DEBUG_BUF_SIZE];
+
+    if( NULL == ssl              ||
+        NULL == ssl->conf        ||
+        NULL == ssl->conf->f_dbg ||
+        level > debug_threshold )
+    {
+        return;
+    }
+
+
+    mbedtls_snprintf( str, sizeof( str ),
+            "print ticket_flags (0x%0x)\n", flag );
+    debug_send_line( ssl, level, file, line, str );
+
+    if ( flag & MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_PSK_RESUMPTION )
+    {
+        mbedtls_snprintf( str, sizeof( str ), "- ALLOW_PSK_RESUMPTION is set.\n" );
+        debug_send_line( ssl, level, file, line, str );
+    }
+    if ( flag & MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_PSK_EPHEMERAL_RESUMPTION )
```

Comment:
```suggestion
   else if ( flag & MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_PSK_EPHEMERAL_RESUMPTION )
```

Reply:
I'm not clear why `else` should be used here. The flags are not to be used exclusively, are they? That is, for example, ALLOW_PSK_EPHEMERAL_RESUMPTION and ALLOW_EARLY_DATA might be set at the same time, and I mean to print all the flags set in `ticket_flags`.

Reply:
Sorry, I misunderstand your idea. 
Seems the code is duplicated in 3 parts. how about combining the 3 **if** together?
such as; define one string array map_str, and use `for `as one example:
for(i = 0; i < 4; i++)
{
        mbedtls_snprintf( str, sizeof( str ), flag& 1<<i ? map_str[ 1<< i] ):"");
        debug_send_line( ssl, level, file, line, str );
}

Reply:
That would be a good idea. I will change the implementation.